parsers/yacc.c

```diff
@@ -52,12 +54,20 @@ static  void change_section (const char *line CTAGS_ATTR_UNUSED,
 		c_source_start = getSourceLineNumber();
 
 		/* Skip the lines for finding the EOF. */
-		while (readLineFromInputFile ())
-			;
+		endCharOffset = 0;
+		last = NULL;
+		while ((tmp = readLineFromInputFile ()))
+			last = tmp;
+		if (last)
+			endCharOffset = strlen ((const char *)last);
+		/* If `last' is too long, strlen returns a too large value
+		   for the positive area of `endCharOffset'. */
```

Comment:
endCharOffset is passed to `mio_seek` as the second argument:
```
int mio_seek (MIO *mio, long offset, int whence);
```

Though it is long, not int, but mio_seek takes the signed value anyway.
So a cast operation from size_to to long is needed. Do you think doing it here is wrong?

I cannot image `last` has such length, but theoretical it has.